### PR TITLE
chore(onboarding): update logging for messaging integration onboarding

### PR DESCRIPTION
**Pull request by ameliahsu**
biggest change is updating the `view` prop of `analyticsParam` to be the page that the user came from (alert creation or project creation) instead of a generic `messaging_integration_onboarding` view

also removed unnecessary analytics that we won't be tracking anymore (ex. modal rendered)

### CODE REVIEW

**Review by Christinarlong — APPROVED** [review]
kewlio

**Review by natemoo-re — APPROVED** [review]
Looks good to me!

File: static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx [hunk 1]
```diff
@@ -9,28 +9,26 @@ import type {
...
 import {useApiQueries, useApiQuery} from 'sentry/utils/queryClient';
 import useOrganization from 'sentry/utils/useOrganization';
 import MessagingIntegrationModal from 'sentry/views/alerts/rules/issue/messagingIntegrationModal';
 
 export enum MessagingIntegrationAnalyticsView {
-  ALERT_RULE_CREATION = 'alert_rule_creation',
-  PROJECT_CREATION = 'project_creation',
+  ALERT_RULE_CREATION = 'alert_rule_creation_messaging_integration_onboarding',
+  PROJECT_CREATION = 'project_creation_messaging_integration_onboarding',
+  DEFAULT = 'messaging_integration_onboarding',
```
Comment (saponifi3d) [inline static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx:20]: is this being used anywhere? I was curious if it was being used as the default for naming, then i didn't see it anywhere else in the PR 😅

File: static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx [hunk 2]
```diff
@@ -9,28 +9,26 @@ import type {
...
-  };
+  analyticsView: MessagingIntegrationAnalyticsView;
   projectId?: string;
   refetchConfigs?: () => void;
 };
 
 function SetupMessagingIntegrationButton({
   refetchConfigs,
-  analyticsParams,
+  analyticsView,
```
Comment (saponifi3d) [inline static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx:31]: If that's the default value, could we use it here as the default? Not sure if we _always_ want to use that as the default though (not enough context)
```suggestion
  analyticsView = MessagingIntegrationAnalyticsView.DEFAULT,
```

File: static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx (hunk 1 above)
Comment (ameliahsu) [reply static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx:20]: oh oops i added the default as a janky fix for something but then i didn't end up needing it 😅  thanks for catching this!

**Pull request merged**